test/default_style.jl

```diff
@@ -4810,4 +4810,23 @@ some_function(
         @test fmt(str_, 2, length(str_) - 1) == str
         @test fmt(str, 2, length(str_)) == str_
     end
+
+    @testset "issue 262 - removal of @ in nested macrocall" begin
+        str = raw":($(@__MODULE__).@macro)"
+        @test fmt(str) == str
+
+        str = raw":($(@__MODULE__).property)"
+        @test fmt(str) == str
+
+        str = raw":($(@__MODULE__))"
+        @test fmt(str) == str
+
+        str_ = raw":($(@__MODULE__).@field.macro)"
+        str = raw":($(@__MODULE__).field.@macro)"
+        @test fmt(str_) == str
```

Comment:
Huh, interestingly, `a.@b.c` is not the same as `@a.b.c` and `a.b.@c`.

```julia
julia> :(@a.b.c)
:(#= REPL[1]:1 =# a.b.@c)

julia> :(a.@b.c)
:(#= REPL[2]:1 =# a.b.var"@c")

julia> :(a.b.@c)
:(#= REPL[3]:1 =# a.b.@c)
```

Although it looks like `a.@b.c` is a syntax (or rather lowering) error:

```julia
julia> Meta.@lower a.@b.c
ERROR: LoadError: syntax: invalid syntax "a.b.@c" around REPL[8]:1
Stacktrace:
 [1] top-level scope
   @ REPL[8]:1
 [2] lower(m::Module, x::Any)
   @ Base.Meta ./meta.jl:124
 [3] top-level scope
   @ REPL[8]:1
in expression starting at REPL[8]:1
```

Not sure what should happen though.  Maybe the safest option is to leave it as-is?

But, if it's easy to transform this way (to avoid special casing or something), personally I think it's OK to transform to the "correct" macro invocation syntax.

Reply:
odd corner case but I've seem to have gotten around it.